src/iceberg/CMakeLists.txt

```diff
@@ -21,7 +21,11 @@ add_iceberg_lib(iceberg
                 SOURCES
                 ${ICEBERG_SOURCES}
                 PRIVATE_INCLUDES
-                ${ICEBERG_INCLUDES})
+                ${ICEBERG_INCLUDES}
+                SHARED_LINK_LIBS
+                IcebergC::iceberg_c_shared
```

Comment:
Do we need to make `iceberg` library mainly interface-oriented and add a new `iceberg-rust-cxx-binding` library to plug in the implementation? It is still unclear to me the extent of plug-ability in this approach.

Reply:
It depends on if we want to use Rust definitions for things like data types too, or if we just want to use it to implement talking to catalogs etc.

Reply:
I think that leveraging rust library is a good approach to move fast. However, there are still users that don't have a rust toolchain, or do not want to add extra dependencies introduced by the rust library.